rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -65,12 +65,19 @@ get_node_names(
     RMW_SET_ERROR_MSG("failed to get default participant qos");
     return RMW_RET_ERROR;
   }
-  node_names->data[0] = rcutils_strdup(participant_qos.participant_name.name, allocator);
-  if (!node_names->data[0]) {
+
+  rcutils_string_array_t tmp_names_list = rcutils_get_zero_initialized_string_array();
+  rcutils_ret = rcutils_string_array_init(&tmp_names_list, length, &allocator);
```

Comment:
I know this might be missing in other parts of our code base, but we should check here that this returns `RCUTILS_RET_OK` (as is done [here](https://github.com/ros2/rmw_connext/pull/303/files#diff-bccfea1954134a50cf75b01ae95188e3R57))

Reply:
fixed